File: apps/mobile/iosApp/Piro/Alerts/AlertsViewModel.swift
import Foundation
import Shared

/// Loads the active alerts for the Alerts tab — the SwiftUI counterpart of the Android `AlertsViewModel`.
@MainActor
final class AlertsViewModel: ObservableObject {
    @Published var loading = true
    @Published var alerts: [AlertDetail] = []
    @Published var error: String?

    private let api: PiroApiClient

    init(api: PiroApiClient) {
        self.api = api
    }

    func refresh() async {
        loading = true
        error = nil
        do {
            alerts = try await api.getAlerts()
        } catch {
            self.error = PiroError.message(error, networkFallback: "Could not load alerts.")
        }
        loading = false
    }
}

File: apps/mobile/androidApp/src/main/kotlin/co/heva/piro/android/alert/AlertsViewModel.kt
package co.heva.piro.android.alert

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import co.heva.piro.shared.api.PiroApiClient
import co.heva.piro.shared.model.AlertDetail
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.flow.update
import kotlinx.coroutines.launch

data class AlertsUiState(
    val loading: Boolean = true,
    val alerts: List<AlertDetail> = emptyList(),
    val error: String? = null,
)

/** Loads the active alerts list for the Alerts tab. */
class AlertsViewModel(private val api: PiroApiClient) : ViewModel() {

    private val _state = MutableStateFlow(AlertsUiState())
    val state: StateFlow<AlertsUiState> = _state.asStateFlow()

    init {
        refresh()
    }

    fun refresh() {
        viewModelScope.launch {
            _state.update { it.copy(loading = true, error = null) }
            try {
                _state.update { it.copy(loading = false, alerts = api.getAlerts()) }
            } catch (e: Exception) {
                _state.update { it.copy(loading = false, error = "Could not load alerts.") }
            }
        }
    }
}
